src/features/collectables/AssertLocationScreen.tsx

```diff
@@ -0,0 +1,656 @@
+
+  const handleOnSearch = useCallback(async () => {
+    if (searchValue) {
+      try {
+        const coords = await forwardGeo.execute(searchValue)
+
+        if (camera?.current && coords) {
+          camera.current.setCamera({
+            animationDuration: 500,
+            centerCoordinate: coords,
+            zoomLevel: MAX_MAP_ZOOM / 1.2,
+          })
+        }
+      } catch (error) {
+        const { message = '' } = error as Error
+        if (message === t('noData')) {
+          await showOKAlert({
+            title: t('generic.error'),
+            message: t('assertLocationScreen.locationNotFound'),
+          })
+        }
+      }
+    }
+
+    hideSearch()
+  }, [camera, t, hideSearch, searchValue, forwardGeo, showOKAlert])
+
+  const onCameraChanged = debounce(async (state: MapboxGL.MapState) => {
+    const { center } = state.properties
+    if (JSON.stringify(center) !== JSON.stringify(mapCenter)) {
+      setMapCenter(center)
+      setTransactionError(undefined)
+      hideSearch()
+    }
+  }, 600)
+
+  const handleUserLocationPress = useCallback(() => {
+    if (camera?.current && userLocation?.current?.state.coordinates) {
+      camera.current.setCamera({
+        animationDuration: 500,
+        zoomLevel: MAX_MAP_ZOOM,
+        centerCoordinate: userLocation.current.state.coordinates,
+      })
+    }
+  }, [userLocation, camera])
+
+  const assertLocation = useCallback(
+    async (type: HotspotType) => {
+      if (mapCenter && entityKey) {
+        setTransactionError(undefined)
+        setAsserting(true)
+        try {
+          hideElevGain()
+          await submitUpdateEntityInfo({
+            type,
+            entityKey,
+            lng: mapCenter[0],
+            lat: mapCenter[1],
+            elevation,
+            decimalGain: gain,
+          })
+          setAsserting(false)
+        } catch (error) {
+          setAsserting(false)
+          Logger.error(error)
+          setTransactionError((error as Error).message)
+        }
+      }
+    },
+    [
+      entityKey,
+      mapCenter,
+      elevation,
+      gain,
+      hideElevGain,
+      setAsserting,
+      setTransactionError,
+      submitUpdateEntityInfo,
+      // nav,
+    ],
+  )
+
+  const handleAssertLocationPress = useCallback(async () => {
+    if (!elevGainVisible) {
+      Alert.alert(
+        t('assertLocationScreen.title'),
+        t('assertLocationScreen.whichLocation'),
+        [
+          {
+            text: 'Iot',
+            onPress: () => setElevGainVisible(true),
+          },
+          {
+            text: 'Mobile',
+            onPress: async () => assertLocation('mobile'),
+          },
+          {
+            text: t('generic.cancel'),
+            style: 'destructive',
+          },
+        ],
+      )
+    } else {
+      // elevGainVisible
+      // we can assume user is asserting location from elevGain UI
+      await assertLocation('iot')
```

Comment:
Should we navigate back to the hotspot details screen when successful?

Reply:
Solid UX question. I think the visual cue of the dot being updated on the map is nice feedback. Might lose that if we navigate back to the details screen? But maybe if we display the location on the details screen that would be good feedback too.